pkg/controllers/csi/driver/volumes/app/publisher_test.go

```diff
@@ -327,19 +328,43 @@ func newPublisherForTesting(mounter *mount.FakeMounter) AppVolumePublisher {
 
 func mockPublishedVolume(t *testing.T, publisher *AppVolumePublisher) {
 	mockUrlDynakubeMetadata(t, publisher)
-	err := publisher.db.InsertVolume(context.Background(), metadata.NewVolume(testVolumeId, testPodUID, testAgentVersion, testTenantUUID, 0))
+
+	mockAppMount := metadata.AppMount{
+		VolumeMeta:        metadata.VolumeMeta{ID: testVolumeId, PodUid: testPodUID},
+		CodeModule:        metadata.CodeModule{Version: testAgentVersion},
+		VolumeMetaID:      testVolumeId,
+		CodeModuleVersion: testAgentVersion,
+		MountAttempts:     0,
+		Location:          publisher.path.AgentRunDirForVolume(testTenantUUID, testVolumeId),
+	}
+
+	err := publisher.db.CreateAppMount(context.Background(), &mockAppMount)
 	require.NoError(t, err)
 	agentsVersionsMetric.WithLabelValues(testAgentVersion).Inc()
 }
 
 func mockFailedPublishedVolume(t *testing.T, publisher *AppVolumePublisher) {
 	mockUrlDynakubeMetadata(t, publisher)
-	err := publisher.db.InsertVolume(context.Background(), metadata.NewVolume(testVolumeId, testPodUID, testAgentVersion, testTenantUUID, dynatracev1beta1.DefaultMaxFailedCsiMountAttempts+1))
+
+	appMount := &metadata.AppMount{
+		VolumeMeta:        metadata.VolumeMeta{ID: testVolumeId, PodUid: testPodUID},
+		CodeModuleVersion: testAgentVersion,
+		MountAttempts:     dynatracev1beta1.DefaultMaxFailedCsiMountAttempts + 1,
+		VolumeMetaID:      testVolumeId,
+	}
+
+	err := publisher.db.CreateAppMount(context.Background(), appMount)
 	require.NoError(t, err)
 }
 
 func mockUrlDynakubeMetadata(t *testing.T, publisher *AppVolumePublisher) {
-	err := publisher.db.InsertDynakube(context.Background(), metadata.NewDynakube(testDynakubeName, testTenantUUID, testAgentVersion, "", 0))
+	tenantConfig := metadata.TenantConfig{
+		Name:                        testDynakubeName,
+		TenantUUID:                  testTenantUUID,
+		DownloadedCodeModuleVersion: testAgentVersion,
+		MaxFailedMountAttempts:      0,
+	}
+	err := publisher.db.CreateTenantConfig(context.Background(), &tenantConfig)
```

Comment:
I think passing context is not a good idea, better to move it to `db`. Also I prefer to have similar method like `metadata.NewTenantConfig`, so we can reuse it in every test.

Reply:
Good idea :+1: Will fix it

Reply:
Done, I added the context to the db struct and now it simplified soooo much the interface and removed a lot of context arguments.